### PR TITLE
Update/in 229

### PR DESCRIPTION
**Pull request by n3op2**
#### What
Mostly addressing comments from previous PR -> https://github.com/digicatapult/inteli-api/pull/15

### Change
- logger: no longer need to pass request object
- logger: a little change the way message being passed to the sub class

### CODE REVIEW

File: app/api-v1/controllers/Recipe/index.js [hunk 1]
```diff
@@ -51,7 +66,7 @@ module.exports = {
         ],
       }
       const token = await runProcess(payload, req.token)
-      const transaction = await client
+      const transaction = await db.client
```
Comment (jonmattgray) [inline app/api-v1/controllers/Recipe/index.js:69]: the response doesn't match any in the API spec such as https://github.com/digicatapult/inteli-api/blob/a96a90a32fb17ce342cac4becaf787b3ec625d62/app/api-v1/api-doc.js#L306

Comment (n3op2) [reply app/api-v1/controllers/Recipe/index.js:56]: updated

File: app/api-v1/routes/recipe/{id}/creation.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ module.exports = function () {
         name: 'id',
         allowEmptyValue: false,
         schema: {
-          $ref: '#/components/schemas/ObjectReference',
+          $ref: '#/components/schemas/ChainAction',
```
Comment (mattdean-digicatapult) [inline app/api-v1/routes/recipe/{id}/creation.js:73]: this looks wrong. I really think this should be an ObjectReference. A `ChainAction` is an object

File: package-lock.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@digicatapult/inteli-api",
-  "version": "1.20.1",
+  "version": "1.20.2",
```
Comment (mattdean-digicatapult) [inline package-lock.json:3]: please don't update package-locks by hand use `npm install`. You've missed a changeon ln 9 btw

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
small changes. I assume token format changes will come later?

File: package-lock.json (hunk 3 above)
Comment (n3op2) [reply package-lock.json:3]: ok. though there was a better way

**Comment by jonmattgray** [comment]
not sure if part of this PR but we surely need an integration test for `POST /recipe/{id}/creation`?

File: app/api-v1/controllers/Recipe/index.js [hunk 4]
```diff
@@ -68,11 +52,10 @@ module.exports = {
...
         ],
       }
-      const token = await runProcess(payload, req.token)
+      runProcess(payload, req.token)
       const transaction = await db.client
         .from('recipe_transactions')
         .insert({
-          token_id: token[0],
           recipe_id: id,
           status: 'Submitted',
```
Comment (jonmattgray) [inline app/api-v1/controllers/Recipe/index.js:60]: missing `type: 'Creation'`

**Review by mattdean-digicatapult — DISMISSED** [review]

File: app/api-v1/controllers/Recipe/index.js (hunk 4 above)
Comment (mattdean-digicatapult) [reply app/api-v1/controllers/Recipe/index.js:60]: well spotted

Comment (n3op2) [reply app/api-v1/controllers/Recipe/index.js:60]: nice one!

Comment (n3op2) [reply app/api-v1/controllers/Recipe/index.js:60]: need to update tests as well

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**